### ISSUE TITLE
worker support?

### ISSUE THREAD

**Issue by horatio-sans-serif** [gofullstack/gearman-node#1]
Are you planning to add support for Node.js based workers?

**Comment by smith** [comment]
Absolutely. That's the next step after I'm done working on the client. Pull requests are always welcome.

**Comment by horatio-sans-serif** [comment]
ok .. I'll take a look into this.

**Comment by smith** [comment]
I pushed to a feature branch with a little readme work on a proposed API: https://github.com/cramerdev/gearman-node/commit/5d345898ccda3784105cc3e6ece4c87d0830b71c

I'd like to use the native EventEmitter API as closely as possible, but I haven't quite worked out exactly how to implement things like RESET_ABILITIES with that.

This Issue can stay open until there's a working implementation.

**Comment by horatio-sans-serif** [comment]
I decided it would be fun to write a full Gearman client and worker in CoffeeScript. So I did that today instead of forking your project. Sorry that I was a tease!

**Comment by smith** [comment]
You're still a tease until you link to the code ;) 

I debated using CS originally, and I'm still thinking about a port. It's not too late for a rewrite, and the goal is to have a working node implementation, so the language doesn't matter.

**Comment by cjoudrey** [comment]
Any updates on a gearman worker?

Also, the proposed API would not properly work for async jobs.

We should probably have a callback argument which we can call to return the payload.

The callback would be in the form of `function(err, data)` like other Node callbacks.

For instance, the reverse worker would become:

``` javascript
worker.on("test", function (data, done) {
    done(null, data.toString("utf8").split("").reverse().join(""));
});
```

Thoughts?

**Comment by perezd** [comment]
Any progress on this? Its fairly useless without a worker API.

**Comment by cjoudrey** [comment]
@perezd Project seems dead :S

**Comment by smith** [comment]
Project is not dead (or useless IMO.) Just haven't had time to work on it much lately. Sorry, but that's just what's going on.

Pull requests are welcome, and I hope to get around to making updates at some point.

**Comment by aredridel** [comment]
+1 on this from me

**Comment by smith** [comment]
https://github.com/andris9/node-gearman does support workers, though there are no plans on that for some other features: background jobs, priorities, status updates, etc.

**Comment by mreinstein** [comment]
Full protocol support for clients and workers  https://github.com/mreinstein/node-gearman
